casepro/msgs/models.py

```diff
@@ -275,14 +282,19 @@ def search(cls, org, user, search):
         if folder == MessageFolder.flagged:
             queryset = queryset.filter(is_flagged=True)
 
-        # archived messages can be implicitly or explicitly included depending on folder
-        if folder == MessageFolder.archived:
-            queryset = queryset.filter(is_archived=True)
-        elif folder == MessageFolder.flagged:
-            if not include_archived:
-                queryset = queryset.filter(is_archived=False)
+        # if this is a refresh we want everything with new actions
+        if last_refresh:
+            queryset = queryset.filter(actions__created_on__gt=last_refresh) |\
+                       queryset.filter(last_action__gt=last_refresh)
         else:
-            queryset = queryset.filter(is_archived=False)
+            # archived messages can be implicitly or explicitly included depending on folder
+            if folder == MessageFolder.archived:
+                queryset = queryset.filter(is_archived=True)
+            elif folder == MessageFolder.flagged:
+                if not include_archived:
+                    queryset = queryset.filter(is_archived=False)
+            else:
+                queryset = queryset.filter(is_archived=False)
```

Comment:
No I don't think so, we need to query all folders for updates, the message might have moved in or out of the folder we are currently looking at.